docs/operate/snapshot-repository.mdx

```diff
@@ -0,0 +1,51 @@
+</Admonition>
+
+For S3 bucket destinations, Restate will use the AWS credentials available from the environment, or the configuration profile specified by `AWS_PROFILE` environment variable, falling back to the default AWS profile.
+
+## Log trimming and snapshots
+
+In a distributed environment, the shared log is the mechanism for replicating partition state among nodes. Therefore it is critical to ensure that all cluster members can get all the relevant updates. This requirement is at odds with an immutable log growing unboundedly. Snapshots enable log trimming in clusters. When partition processors successfully publish a snapshot, they update their "archived" log sequence number (LSN). This reflects the position in the log at which the snapshot was taken.
+
+By default, Restate will attempt to trim logs once an hour which you can override or disable in the server configuration:
+
+```toml
+[admin]
+log-trim-interval = "1h"
+```
+
+This interval is only the check and not a guarantee that logs will be trimmed. Restate will automatically determine the appropriate safe trim point for each partition's log.
+
+If replicated logs are in use in a clustered environment, the log safe trim point will be determined based on the archived LSN. If a snapshot repository is not configured, then archived LSNs are not reported. Instead, the safe trim point will be determined by the smallest reported persisted LSN across all known processors for the given partition. Single-node local-only logs are also trimmed based on the partitions' persisted LSNs.
+
+The presence of any dead nodes in a cluster will cause trimming to be suspended for all partitions, unless a snapshot repository is configured. This is because we can not know what partitions may reside on the unreachable nodes, which will become stuck when the node comes back.
+
+When a node starts up with pre-existing partition state and finds that the partition's log has been trimmed to a point beyond the most recent locally-applied LSN, the node will attempt to download the latest snapshot from the configured repository. If a suitable snapshot is available, the processor will re-bootstrap its local state and resume applying the log.
+
+<Admonition type="note" title="Handling log trim gap startup errors">
+    If you observe repeated `Shutting partition processor down because it encountered a trim gap in the log.` errors in the Restate server log, that is an indication that a processor is failing to start up due to missing log records. In order to record, you must ensure that a snapshot repository is correctly configured and accessible from the the node experiencing errors. If no snapshots were taken previously, you will need to configure another node which has the necessary state, to first publish a snapshot for the affected partition(s).
```

Comment:
"In order to resolve this failure, you must ..."?

Reply:
I think I meant to write "recover" there 😅 Yours reads better, thank you!